tests/network/vmi_multus.go

```diff
@@ -1283,6 +1283,8 @@ var _ = SIGDescribe("[Serial]Macvtap", func() {
 
 				serverVMI, err = createFedoraVMIRandomNode(macvtapNetworkName, macAddress)
 				Expect(err).NotTo(HaveOccurred(), "must have succeeded creating a fedora VMI on a random node")
+				// Ensure QEMU GuestAgent is up as it should report network interfaces stats
+				tests.WaitAgentConnected(virtClient, serverVMI)
 				Expect(serverVMI.Status.Interfaces).NotTo(BeEmpty(), "a migrate-able VMI must have network interfaces")
```

Comment:
you might need to add here Eventually,
because the virt-launcher starts polling when it gets the agent connected event,
and until the response does all its way to virt-handler, the test, which runs on other process can reach this line and find this line empty.

Reply:
The VMI Status.Interfaces wont be empty at this stage as the VMI is Ready
I guess I should move this line to be after assertion on it

Reply:
> > Note - Waiting for the agent connected event, doesnt say anything at all about if the data is updated or not, it just say that the agent is connected (and its also the trigger to start polling the data by virt-launcher).
> 
> In practice it seems that the polling stars as soon as the condition appears, then in turn the VMI interfaces details are updated.

We had races as such, and fixed them, the practice is different when there is a race.
I am not saying that there is a mandatory AI atm, but felt its right to mention it.

Reply:
> The VMI Status.Interfaces wont be empty at this stage as the VMI is Ready

Maybe this is what happens usually but its not mandatory,
The fact the VM is running and guest agent is running, doesn't mean the NM finished running.
The guest agent might send an empty report (saw it), and when there is a guest agent, it is the one that dictates the interfaces.

Again i am not saying there is a mandatory AI here, but trying just to be accurate.

Reply:
Note that we already perform a polling on the macvtap interface IP address
https://github.com/kubevirt/kubevirt/blob/b5ec3eb68580249567db994f1df207148345bb16/tests/network/vmi_multus.go#L1243
https://github.com/kubevirt/kubevirt/blob/b5ec3eb68580249567db994f1df207148345bb16/tests/network/vmi_multus.go#L1284-L1292
So instead of increasing the timeout arbitrarily, its better to make sure there will be a report on time by waiting for the AgentConnected condition.